src/map/WorldMap.kt

```diff
@@ -42,9 +17,38 @@ interface WorldMap {
     fun positionFor(mapObject: MapObject): Coordinate
     fun getNode(positionFor: Coordinate): Node
     fun moveObject(actor: MapObject, vector: Vector)
+    fun getAllCoordinates(): Set<Coordinate>
+    fun getSurrounding(actor: Actor): WorldMap
 }
 
-class CoordinateNodeWorldMap(private val nodes: MutableMap<Coordinate, Node>) : WorldMap {
+class Memory(private val nodes: MutableMap<Coordinate, Node>) {
+
+    companion object {
+        fun init(coordinates: Set<Coordinate>): Memory {
```

Comment:
Naming it something like `new` might be better to relay that this is actually a constructor.
Alternatively, I prefer implementing it as a top level function with the same name as the class, because it is more discoverable. Unless companion object is strictly necessary, which is rare.

Reply:
https://stackoverflow.com/questions/19299525/kotlin-secondary-constructor

Reply:
you already fixed this for me